### PR TITLE
[FIX] avoid calls to remote servers when saving streamdetails for dvd…

### PR DESCRIPTION
**Pull request by the-black-eagle**
…/bluray

## Description
Current implementation uses a call to `file.IsBluray()` which in turn calls `GetOpticalMediaPath()` if the file is not bluray related. This checks for the existence of various files.  If the file is a stream, this causes 6 calls to the remote server. This can be avoided by calling `URIUtils::IsBluray()` directly as at this point any bluray media already has the correct protocol added and the file existence checks are not needed.
The issue can be easily replicated by playing a `.strm` file containing a url to a remotely streamed video. Curl logging will show 6 calls to the remote server.

## Motivation and context
Kodi should not probe remote servers for the existence of files when playing a stream where there are no stream details in the db. Issue was brought to light by @matthuisman in https://github.com/xbmc/xbmc/pull/19776  We do still want to save the details for bluray/dvd/iso material if there are no existing details.

## How has this been tested?
Tested against a local and remote http server, with bluray images and directories on the local server and a stream from the remote server.  Curl logging enabled.  Details were correctly saved for all bluray items, no details were saved for the stream and no attempts were made to test for the existence of files.  Also checked that dvd material and iso's still work correctly.

## What is the effect on users?
No impact on end users.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/Application.cpp [hunk 1]
```diff
@@ -2991,12 +2991,9 @@ void CApplication::OnPlayBackStarted(const CFileItem &file)
   // check if VideoPlayer should set file item stream details from its current streams
   if (file.GetProperty("get_stream_details_from_player").asBoolean()
       || ((!file.HasVideoInfoTag() || !file.GetVideoInfoTag()->HasStreamDetails())
-      && (file.IsDiscImage()
+      && (URIUtils::IsBluray(file.GetPath())
       || file.IsDVDFile()
-      || file.IsBDFile()
-      || file.IsBluray()
-      || file.IsDVD()
-      || file.IsOnDVD())))
```
Comment (arnova) [inline xbmc/Application.cpp:2999]: Why did you remove isDVD & IsOnDVD?

Comment (the-black-eagle) [reply xbmc/Application.cpp:2999]: Those checks aren't needed. If the file is on a DVD (either an actual physical one or on disk in a video_ts directory) then it either has to be a `.vob` file or an `.ifo`. They are the only playable files that can be in/on that media and `IsDVDFile()` will return true for them.
I don't see the point of checking if it's an actual physical DVD or that the file is on one when that is irrelevant.  All we need to know is that it is either `video_ts.ifo`, starts with `vts_` and either ends with `.vob` or `_0.ifo` or is `video_ts.vob`.

As that is all you can have on a dvd (we don't need to worry about the .bup files) then the check for `IsDVDFile()` is quite sufficient to identify dvd media regardless of the actual physical source, which we don't need for this case.

If the dvd is an `.iso` then that test will not detect it as such so `IsDiscImage()` is still needed for those cases.  However, if the disc image is a bluray, Kodi will have already identified it as such and the correct protocol will be present - if you have it enabled, the simplified bluray menu dialog will already be on screen before we get to these tests.

**Comment by the-black-eagle** [comment]
If nobody has any objections, I'll merge this tomorrow.

**Comment by puenktchen** [comment]
@the-black-eagle can you please backport this or at least provide a Matrix testbuild for the time being? This fix is highly needed! Thanks.
